### PR TITLE
Initsync skip local blobs

### PR DESCRIPTION
**Pull request by kasey**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**

Modify blob request methods to skip slot ranges where we already have blobs on disk. Also modifies the availability store so that if a blob is visible in the pruner cache, it does not require that index to be in the availability store cache, only verifying and savings blobs that are not yet known to the cache.

### CODE REVIEW

File: beacon-chain/das/availability.go [hunk 1]
```diff
@@ -94,6 +94,11 @@ func (s *LazilyPersistentStore) IsDataAvailable(ctx context.Context, current pri
 	entry := s.cache.ensure(key)
 	defer s.cache.delete(key)
 	root := b.Root()
+	sumz, err := s.store.WaitForSummarizer(ctx)
+	if err == nil {
+		entry.setDiskSummary(sumz.Summary(root))
+	}
+
```
Comment (prestonvanloon) [inline beacon-chain/das/availability.go:101]: Please do something about err != nil

File: beacon-chain/sync/initial-sync/blocks_fetcher.go [hunk 2]
```diff
@@ -397,76 +392,127 @@ func lowestSlotNeedsBlob(retentionStart primitives.Slot, bwb []blocks2.BlockWith
...
+		// We found a missing blob
+		needed := &blobRange{low: hci.slot, high: hci.slot}
+		for z := len(fc) - 1; z > i; z-- {
+			hcz := fc[z]
+			if bs.Summary(hcz.root).AllAvailable(hcz.count) {
+				continue
+			}
+			needed.high = hcz.slot
+			return needed
+		}
```
Comment (prestonvanloon) [inline beacon-chain/sync/initial-sync/blocks_fetcher.go:436]: Does this assume that fc is a sorted list? If so, is that a safe assumption?

Comment (kasey) [reply beacon-chain/sync/initial-sync/blocks_fetcher.go:436]: It's a requirement per specs: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/sync/rpc_send_request.go#L93-L103

File: beacon-chain/das/availability.go (hunk 1 above)
Comment (kasey) [reply beacon-chain/das/availability.go:101]: The summarizer is an optional optimization. If for whatever reason the summarizer didn't successfully init, we just don't use it. But a debug log would make sense.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**